flatliners/alertcomparisonscore.py

```diff
@@ -5,14 +5,16 @@
 from dataclasses import dataclass
 import flatliners
 
+from cachetools import LRUCache
+
 class AlertComparisonScore(BaseFlatliner):
-    def __init__(self):
+    def __init__(self, max_cache_size: int = 500):
         super().__init__()
 
-        self.score = dict()
-        self.clusters = dict()
-        self.versions = dict()
-        self.alert_deltas = dict()
+        self.score = LRUCache(maxsize=max_cache_size)
```

Comment:
what happens if we exceed the cache_size?

Reply:
It will still add new items, but it will remove the least recently used item so the number of items in the cache remains the same